src/main/java/seedu/address/logic/parser/ArgumentMultimap.java

```diff
@@ -57,4 +59,28 @@ public void put(Prefix prefix, String argValue) {
     public String getPreamble() {
         return getValue(new Prefix("")).orElse("");
     }
+
+    /**
+     * Returns true if and only if the prefix queried has exactly one value.
+     */
+    public boolean hasExactlyOneValue(Prefix queryPrefix) {
+        return getAllValues(queryPrefix).size() == 1;
+    }
+
+    /**
+     * Returns true if and only if exactly one prefix has non-empty value among all queried prefixes.
+     */
+    public boolean hasExactlyOneQueriedPrefix(Prefix... queryPrefixes) {
+        return getNonEmptyPrefixes(queryPrefixes).size() == 1;
+    }
+
+    /**
+     * Returns a list of prefixes that have values, among all the queried prefixes.
+     *
+     * @param prefixes The list of prefixes to query
+     */
+    public List<Prefix> getNonEmptyPrefixes(Prefix... prefixes) {
+        return Stream.of(prefixes).filter(prefix -> getValue(prefix).isPresent())
+                .collect(Collectors.toList());
+    }
```

Comment:
Good use of varg which makes the method flexible to a number of arguments! Good matching with stream operations to return the list of prefix.